iceoryx_hoofs/source/file_reader/file_reader.cpp

```diff
@@ -33,34 +35,31 @@ FileReader::FileReader(const std::string& f_fileName, const std::string& f_fileP
 
     if (!isOpen())
     {
-        errorHandler(Error::kFILEREADER__FAILED_TO_OPEN_FILE, [=]() {
-            switch (m_errorMode)
-            {
-            case ErrorMode::Ignore:
-            {
-                break;
-            }
-            default:
-            case ErrorMode::Inform:
-            {
-                std::cerr << "\033[5;31m"
-                          << "Could not open file '" << m_file << "'."
-                          << "\033[0m" << std::endl;
-                break;
-            }
-            case ErrorMode::Terminate:
-            {
-                std::cerr << "\033[5;31m"
-                          << "Could not open file '" << m_file << "'. Exiting!"
-                          << "\033[0m" << std::endl;
-                std::terminate();
-                break;
-            }
-            }
-        });
+        switch (m_errorMode)
```

Comment:
Is semantically (almost) equivalent (the code `Error::kFILEREADER__FAILED_TO_OPEN_FILE` was removed, intentionally I presume to have no such codes in hoofs).

It looks a bit weird though in general, as is kind of invokes its own error handling inside the switch. If everything is generalized properly, the error handler should handle the decision whether to terminate etc. based on a severity level and a safety mode (`ErrorMode` is kind of like this).

The user only should provide the safety/handling mode, error severity and optional custom code/messages. It is important to have the code as concise as possible for this but also used in the same way in all places. 

NB: I am just trying to think through how it could ideally work in the future, nothing to change here for now.